lib/wasi/src/state/mod.rs

```diff
@@ -163,12 +168,111 @@ pub struct WasiFs {
     inode_counter: Cell<u64>,
     /// for fds still open after the file has been deleted
     pub orphan_fds: HashMap<Inode, InodeVal>,
+    #[cfg_attr(feature = "enable-serde", serde(skip, default = "default_fs_backing"))]
+    fs_backing: Box<dyn FileSystem>,
+}
+
+/// Returns the default filesystem backing
+pub(crate) fn default_fs_backing() -> Box<dyn wasmer_vfs::FileSystem> {
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "host_fs")] {
+            Box::new(wasmer_vfs::host_fs::HostFileSystem)
+        } else if #[cfg(feature = "mem_fs")] {
+            Box::new(wasmer_vfs::mem_fs::MemFileSystem::default())
+        } else {
+            Box::new(FallbackFileSystem::default())
+        }
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct FallbackFileSystem;
```

Comment:
`FallbackFileSystem` is not required if at least `host_fs` _or_ `mem_fs` is required. One of them is required. But they are mutually exclusive.

I think it's the expected behavior here. WASI without a FS is… useless…, right?

Reply:
Not at all.. some commands (like cowsay) operate without a filesystem... they just do compute :)